controllers/managedmcg_controller.go

```diff
@@ -194,6 +199,17 @@ func (r *ManagedMCGReconciler) reconcilePhases() (reconcile.Result, error) {
 		if err := r.reconcileOCSCSV(); err != nil {
 			return ctrl.Result{}, err
 		}
+
+		logger := log.FromContext(ctx)
+		instance := configv1.ClusterVersion{}
+		if err := r.Client.Get(context.TODO(), req.NamespacedName, &instance); err != nil {
```

Comment:
IMO, it is always better to fetch the ctx from methods than the saved one. Because the initialized/saved context may become invalid . Also in our code we are not using the ctx to control the flow. 

I will change it suggested way to keep this things matched